include/yaramod/types/regexp.h

```diff
@@ -365,6 +413,11 @@ class Regexp : public String
 		_suffixMods = suffixMods;
 	}
 
+	const std::shared_ptr<RegexpUnit>& getUnit() const { return _unit; }
+
+	void setUnit(const std::shared_ptr<RegexpUnit>& unit) { _unit = unit; }
+	void setUnit(std::shared_ptr<RegexpUnit>&& unit) { _unit = std::move(unit); }
```

Comment:
I am just wondering, there are a lot of such overloaded functions (one taking an l-value ref-to-const, one r-value ref). Since they are in header files, wouldn't it be easier to use a single function taking a forwarding reference and using `std::forward()` instead?
```cpp
template <typename T>
void setUnit(T&& unit) { _unit = std::forward<T>(unit); }
```
Or do you prefer the current approach as it contains explicit types? Or forwarding references don't bode well with Python bindings?

This may not be directly related to the PR as there may be already other places with such overloads in the code, so I am also CCing @metthal.

Reply:
Result of me not knowing this stuff like 4 years ago when I was writing very first version of yaramod. It should be sufficient to have single universal reference method and Python bindigs should cope with it but it'll require explicit specification of what type to infer. These interfaces will most probably change as we'll move towards unique pointers once we get rid of bison and flex so I don't necessarily require from Matej.

Reply:
Thank you for the response! Consider it a tip to improve the code in the future then :slightly_smiling_face: I will mark the conversion as resolved as this is not directly related to the present PR.

Reply:
We appreciate your tips. Will keep that in mind (cc @TadeasKucera).